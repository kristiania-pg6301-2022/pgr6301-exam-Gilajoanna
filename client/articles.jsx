import * as React from "react";
import { Link, Route, Routes } from "react-router-dom";
import { useLoader } from "./globals/useLoader";
import { fetchJSON } from "./globals/fetchJSON";
import { AddNewArticle } from "./addNewArticle";

function ArticleItem({ article: { title, category, content, author } }) {
  return (
    <div>
      <h2>{title}</h2>
      <div>
        <strong>{category}</strong>
      </div>
      <div>
        Author: <i>{author}</i>
      </div>
      <div>{content}</div>
    </div>
  );
}

export function ListArticles({ listAllArticles }) {
  const { loading, data, error } = useLoader(listAllArticles);

  if (loading) {
    return <div>Please wait..</div>;
  }

  if (error) {
    return (
      <div>
        <h1>An error occurred</h1>
        <div id="error-message">Something went wrong. Please try again.</div>
      </div>
    );
  }

  return (
    <div>
      <h1>List of articles</h1>

      {data.map((article) => (
        <ArticleItem key={article.title} article={article} />
      ))}
    </div>
  );
}

export function Articles() {
  async function listAllArticles() {
    return await fetchJSON("/api/articles");
  }

  return (
    <Routes>
      <Route
        path={""}
        element={<ListArticles listAllArticles={listAllArticles} />}
      />
      <Route path={"new"} element={<AddNewArticle />} />
    </Routes>
  );
}
