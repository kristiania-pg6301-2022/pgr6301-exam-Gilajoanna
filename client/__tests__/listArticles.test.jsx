import * as ReactDOM from "react-dom";
import * as React from "react";
import { ListArticles } from "../articles.jsx";

describe("List articles component", () => {
  it("renders ListArticle", () => {
    const element = document.createElement("div");
    ReactDOM.render(<ListArticles />, element);
    expect(element.innerHTML).toMatchSnapshot();
  });
});
