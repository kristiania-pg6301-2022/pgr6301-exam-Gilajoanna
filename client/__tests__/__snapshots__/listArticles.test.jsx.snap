// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `"<div>Please wait..</div>"`;

exports[`List articles component renders ListArticle 1`] = `"<div>Please wait..</div>"`;

exports[`List articles component shows h1 1`] = `"<div>Please wait..</div>"`;

exports[`List articles component shows loading when loading 1`] = `"<div>Please wait..</div>"`;
